app/assets/javascripts/bindingtarifftraderfrontend.js

```diff
@@ -139,6 +145,15 @@ $(document).ready(function () {
         $('.error-summary').focus();
     }
 
+    function updateSessionHistory() {
+        const historyStack = JSON.parse(sessionStorage.getItem("historyStack")) || [];
+        const latestUrl = historyStack[historyStack.length - 1]
+        if (latestUrl !== window.location.href) {
+            historyStack.push(window.location.href)
+            sessionStorage.setItem("historyStack", JSON.stringify(historyStack));
+        }
+    }
```

Comment:
Might be worth thinking about the same kind of issues here too, although I think the effect will be less problematic here as it will just push the current page URL